### PR TITLE
Georgi Shopov Hw01

### PR DESCRIPTION
**Pull request by georgiiishopovv**
Georgi Shopov

### CODE REVIEW

File: Main.hs [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+
+short :: [a] -> Bool
+short []        = True
+short [x]       = True
+short [x, y]    = True
+short [x, y, z] = False
+short (x:y:z:_) = False
+
+lovely :: [Int] -> Bool
+lovely a = (short a) || thirdElement a
```
Comment (kajigor) [inline Main.hs:14]: Try to avoid unnecessary parentheses.

File: Main.hs [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+
+import Control.Monad (unless)
+import Text.Printf (printf)
+
+short :: [a] -> Bool
+short []        = True
+short [x]       = True
+short [x, y]    = True
+short [x, y, z] = False
+short (x:y:z:_) = False
```
Comment (kajigor) [inline Main.hs:11]: It's better to use wildcards when you are not using the names introduced in a pattern match.

**Review by kajigor — CHANGES_REQUESTED** [review]

File: Main.hs [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+short []        = True
+short [_]       = True
+short [_, _]    = True
+short [_, _, _] = False
+short (_:_:_:_) = False
+
+lovely :: [Int] -> Bool
+lovely a = short a || thirdElement a
+   where 
+      thirdElement (_:_:x:_) = if x == 14 then True else False
```
Comment (alex28sh) [inline Main.hs:16]: too much overhead

thirdElement (_:_:14:_) = True - is much easier

File: Main.hs [hunk 4]
```diff
@@ -0,0 +1,198 @@
...
+
+import Control.Monad (unless)
+import Text.Printf (printf)
+
+short :: [a] -> Bool
+short []        = True
+short [_]       = True
+short [_, _]    = True
+short [_, _, _] = False
+short (_:_:_:_) = False
```
Comment (alex28sh) [inline Main.hs:11]: last 2 cases can be united

**Review by alex28sh — COMMENTED** [review]
ok

**Pull request closed without merging**